compiler/frontend/src/org/jetbrains/kotlin/psi/KtPsiUtil.java

```diff
@@ -513,6 +513,16 @@ public static boolean areParenthesesNecessary(
             return false;
         }
 
+        if (innerExpression instanceof KtBinaryExpression) {
+            PsiElement expr = innerExpression.getFirstChild();
+            while (expr != null) {
+                if (expr instanceof PsiWhiteSpace && (expr.textContains('\n') || expr.textContains('\r'))) {
```

Comment:
IntelliJ always normalizes text of files to use \n line breaks, the \r will never be there. Please remove it from the check.